rfcs/20200813-clang-tidy/README.md

```diff
@@ -0,0 +1,196 @@
+# Enable Clang-tidy for oneDNN
```

Comment:
(random place) do we have an idea (at least roughly) when we plan to implement the RFC?

Reply:
I have a branch that supports clang-tidy as proposed. Once this RFC is promoted I'll apply the checks one by one. If a check requires to modify the code then I'll put the changes into a separate commit so that we can clearly see what each check changes in the codebase.

Reply:
Good to know, thank you!
Maybe it is worth mentioning that in RFC, saying we plan the initial support of clang-tidy in around oneDNN v1.7 - v2.0 (so that other could plan their changes accordingly -- I guess there would be quite a few changes to the code base).

Reply:
I put the information about the date in the description to this PR.
This RFC stays open for 1 week to get feedback.